src/track/beats.cpp

```diff
@@ -309,8 +319,10 @@ mixxx::BeatsPointer Beats::fromBeatGridByteArray(
         bpm = mixxx::Bpm(pBlob->bpm);
     }
 
+    u_int8_t downbeatsOffset = grid.has_downbeats_offset() ? grid.downbeats_offset() : 0;
```

Comment:
we seldomnly use `u_int8_t`. Just stick to the plain `int`, especially since the int you store in the beatgrid is actually a 32-bit int.